### DISCUSSION TITLE
How to set up Debugger plugin for python debugging?

### DISCUSSION THREAD

**Discussion by ronx6** [geany/geany#3692] [Q&A] [answered]
The Debugger plugin uses GDB which I read somewhere, has functionality to debug python code. I have a python file called this_script.py how do I debug the code inside this_script.py? I want to debug using step over, step into, step out etc. Can someone give a step by step info of how to set up python debugging in Debugger plugin.

**Comment by elextr** [comment]
https://devguide.python.org/development-tools/gdb/

**Reply by ronx6** [reply]
Thank you for the quick reply but how do I set up this in geany debugger. In vscode I have to click the debug button and select file to go to debugging mode.

In the debugger I have a Target tab and inside we have an option to select target and command line arguments. 
I have added my file this_script.py in the target and clicked run but I got "Error Loading  File" alert.

**Reply by elextr** [reply, marked as answer]
What the linked article makes clear is that:

1. GDB can use extension scripts written in Python
2. some extensions are available to help debug the C code of the Python interpreter while its interpreting other Python scripts

Not that GDB by itself supports debugging python scripts.

**Comment by elextr** [comment]
What you likely want is something like [IDLE](https://docs.python.org/3/library/idle.html)

**Comment by mayanksingh2298** [comment]
We are working to build a super debugger that lets you analyze arbitrary variables in running code without re-deployment. It is an IDE plugin and best of all we are charging $0 for it.

If you find this interesting, you can get started here: https://docs.ctrlb.ai/getting-started-in-2-minutes
